src/cmdline.rs

```diff
@@ -228,6 +237,45 @@ pub fn parse_args() -> Result<Config> {
                         .takes_value(true),
                 ),
         )
+        .subcommand(
+            SubCommand::with_name("list-stream")
+                .about("List available images in a Fedora CoreOS stream")
+                .usage("coreos-installer list-stream [OPTIONS] [<architecture> [<platform>]]")
```

Comment:
I personally find the UX here confusing, as it seems to switch listing different entity types depending on the optional parameters.

I'd probably go for a different arrangement, where the different cases are split apart via flags or subcommands. A strawman could be:
 * `-t`: entity-type to list. Allowed values: `basearch`, `platform`, `format`. Default: all.
 * `-a`: basearch to select for filtering. Default: none (include all arch in results).
 * `-p`: platform to select for filtering. Default: none (include all platforms in results).